### PR TITLE
WEBC-568

### PR DESCRIPTION
**Pull request by awesselburg**

### CODE REVIEW

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/SalesOrderAfter.php [hunk 1]
```diff
@@ -55,11 +55,14 @@ public function execute(Varien_Event_Observer $observer)
...
     /**
-     * @return int|null
-     * @throws Mage_Core_Model_Store_Exception
+     * @return int|false
      */
     protected function getShopgateStoreId()
     {
-        return Mage::app()->getStore(self::SHOPGATE_STORE_CODE)->getId();
+        try {
+            return Mage::app()->getStore(self::SHOPGATE_STORE_CODE)->getId();
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/SalesOrderAfter.php]: My suggestion is to not use a try/catch as it's not a proper way to check for existence of a config. Just load the store, if it is not there, return false.
e.g.
```
$store = Mage::getModel('core/store')->load($storeCode);
if ($store->getId()) {
    //the store exists
}
```

Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/SalesOrderAfter.php]: changed

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/SalesOrderAfter.php [hunk 2]
```diff
@@ -59,10 +59,13 @@ public function execute(Varien_Event_Observer $observer)
...
-        try {
-            return Mage::app()->getStore(self::SHOPGATE_STORE_CODE)->getId();
-        } catch (Mage_Core_Model_Store_Exception $exception) {
-            return false;
+        /** @var Mage_Core_Model_Store $store */
+        $store = Mage::app()->getStore(self::SHOPGATE_STORE_CODE);
+
+        if ($store->getId()) {
+            $store->getId();
         }
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/SalesOrderAfter.php]: Öhm, doesn't make sense at all?!

Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/SalesOrderAfter.php]: return added ^^

**Pull request merged**